packages/flutter/lib/src/material/floating_action_button.dart

```diff
@@ -268,6 +258,18 @@ class _FloatingActionButtonState extends State<FloatingActionButton> {
       child: result,
     );
 
+    if (widget.tooltip != null) {
+      final Widget tooltip = new MergeSemantics(
+        child: new Tooltip(
+          message: widget.tooltip,
+          child: result,
+        ),
+      );
+      // The long-pressable area for the tooltip should always be as big as
+      // the tooltip even if there is no child.
+      result = widget.child != null ? tooltip : new SizedBox.expand(child: tooltip);
```

Comment:
This doesn't look right. A childless FAB's size shouldn't depend on the size of the tooltip's message.

Reply:
I didn't add this behavior, it was added in [pull/16168](https://github.com/flutter/flutter/pull/16168), so I don't have the full context. It's possible that my moving the tooltip wrapper to after the `RawMaterialButton` also fixed this bug. 

I removed this line in a recent commit, since [this test](https://github.com/HansMuller/flutter/blob/da89186cd6dc5496d05c06fcbfada9115702a3d2/packages/flutter/test/material/floating_action_button_test.dart#L72) still passes after the change, do you feel its safe to remove?

Reply:
Put in by me no less (I'd forgotten)!

The problem here is that before the tooltip's size was constrained by the RawMaterialButton, which has a fixed size/shape the case of a FAB. Since the Tooltip now contains the RawMaterialButton, its size isn't constrained anymore.